services/tests/startup.rs

```diff
@@ -0,0 +1,139 @@
+use assert_cmd::prelude::*;
+
+use geoengine_services::test_data;
+use serial_test::serial;
+use std::{
+    io::BufRead,
+    process::{Command, Stdio},
+};
+
+#[cfg(feature = "pro")]
+#[serial]
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn it_starts_postgres_backend_with_no_warnings() {
+    use geoengine_services::util::config::get_config_element;
+
+    const SCHEMA_NAME: &str = "it_starts_postgres_backend_with_no_warnings";
+
+    async fn run_server_and_check_warnings() {
+        let mut server = Command::cargo_bin("main")
+            .unwrap()
+            .env("GEOENGINE_WEB__BACKEND", "postgres")
+            .env("GEOENGINE_POSTGRES__SCHEMA", SCHEMA_NAME)
+            .stderr(Stdio::piped())
+            .spawn()
+            .unwrap();
+
+        let stderr = server.stderr.take().expect("failed to read stderr");
+
+        let mut reader = std::io::BufReader::new(stderr);
+
+        let mut startup_succesful = false;
+        for _ in 0..99 {
+            let mut buf = String::new();
+            reader.read_line(&mut buf).unwrap();
+
+            // println!("{buf}");
+
+            assert!(!buf.contains("WARN"));
+
+            if buf.contains("Tokio runtime found") {
+                startup_succesful = true;
+                break;
+            }
+        }
```

Comment:
Can't you read after you've shutdown the server?

Reply:
Or… use async stderr from tokio:
https://docs.rs/tokio/1.27.0/tokio/io/fn.stderr.html

Reply:
no. How should I know when I can shutdown the server without reading the logs?

why should I use the stderr from tokio? I think it's ok to block here.

Reply:
You are already in an async function, so just await?

Reply:
there is no read_line method in AsyncReadExt, so I would prefer to leave the code like this and not implement it on my own.

Reply:
And what about warnings that occur after the statement `Tokio runtime found`?
Now, you quit reading the 99 lines.

Reply:
This test is only for checking warning during start up. "Tokio runtime found" is the last message during a successful startup. I haven't found a different way to detect this.

I quit after 99 lines because I don't want the test to run indefinitely and we never output even close to that number of messages during a successful stratup.